examples/step-37/doc/results.dox

```diff
@@ -409,3 +412,240 @@ number of threads in the MPI_InitFinalize data structure in the main function,
 and set the MatrixFree::AdditionalData::tasks_parallel_scheme to
 partition_color to actually do the loop in parallel. This use case is
 discussed in step-48.
+
+<h4> Inhomogeneous Dirichlet boundary conditions </h4>
+
+The presented program assumes homogeneous Dirichlet boundary conditions. When
+going to non-homogeneous conditions, the situation is a bit more intricate. To
+understand how to implement such a setting, let us first recall how these
+arise in the mathematical formulation and how they are implemented in a
+matrix-based variant. In essence, an inhomogeneous Dirichlet condition sets
+some of the nodal values in the solution to given values rather than
+determining them through the variational principles,
+@f{eqnarray*}
+u_h(\mathbf{x}) = \sum_{i\in \mathcal N} \varphi_i(\mathbf{x}) u_i =
+\sum_{i\in \mathcal N \setminus \mathcal N_D} \varphi_i(\mathbf{x}) u_i +
+\sum_{i\in \mathcal N_D} \varphi_i(\mathbf{x}) g_i,
+@f}
+where $u_i$ denotes the nodal values of the solution and $\mathcal N$ denotes
+the set of all nodes. The set $\mathcal N_D\subset \mathcal N$ is the subset
+of the nodes that are subject to Dirichlet boundary conditions where the
+solution is forced to equal $u_i = g_i = g(\mathbf{x}_i)$ as the interpolation
+of boundary values on the Dirichlet-constrained node points $i\in \mathcal
+N_D$. In this formula, where we have assumed that the degrees of freedom are
+numbered such that the ones subject to a Dirichlet condition are all appended
+to the end of the list of degrees of freedom. We then insert this solution
+representation into the weak form, e.g. the Laplacian shown above, and move
+the known quantities to the right hand side:
+@f{eqnarray*}
+(\varphi_i, u_h)_\Omega &=& (\varphi_i, f)_\Omega \quad \Rightarrow \\
+\sum_{j\in \mathcal N \setminus \mathcal N_D}(\varphi_i,\varphi_j)_\Omega \, u_j &=&
+(\varphi_i, f)_\Omega
+-\sum_{j\in \mathcal N_D} (\varphi_i,\varphi_j)_\Omega\, g_j.
+@f}
```

Comment:
Don't you want to consider here a Laplace matrix instead of a mass matrix, i.e. 
```
(\nabla \varphi_i, \nabla \u_h) &=& (\varphi_i, f)_\Omega
```
?

Reply:
Of course, thanks!